src/gas/gas_utils.cairo

```diff
@@ -127,17 +173,33 @@ fn estimate_execute_deposit_gas_limit(data_store: IDataStoreDispatcher, deposit:
 fn estimate_execute_withdrawal_gas_limit(
     data_store: IDataStoreDispatcher, withdrawal: Withdrawal
 ) -> u128 {
-    //TODO
-    0
+    let gas_per_swap = data_store.get_u128(keys::single_swap_gas_limit());
+    let swap_count = withdrawal.long_token_swap_path.len() + withdrawal.short_token_swap_path.len();
+    let gas_for_swaps = swap_count.into() * gas_per_swap;
+    return data_store.get_u128(keys::withdrawal_gas_limit_key())
+        + withdrawal.callback_gas_limit
+        + gas_for_swaps;
 }
 
 /// The estimated gas limit for orders.
 /// # Arguments
 /// * `data_store` - The data storage contract dispatcher.
 /// * `order` - The order to estimate the gas limit for.
 fn estimate_execute_order_gas_limit(data_store: IDataStoreDispatcher, order: @Order) -> u128 {
-    // TODO
-    0
+    if (is_increase_order(*order.order_type)) {
+        return estimate_execute_increase_order_gas_limit(data_store, *order);
+    }
+
+    if (is_decrease_order(*order.order_type)) {
+        return estimate_execute_decrease_order_gas_limit(data_store, *order);
+    }
+
+    if (is_swap_order(*order.order_type)) {
+        return estimate_execute_swap_order_gas_limit(data_store, *order);
+    }
+
+    panic(array![OrderError::UNSUPPORTED_ORDER_TYPE]);
+    return 0; //question
```

Comment:
this method originally panics and doesn't return anything, Cairo syntax requires a return but currently it's unreachable.

not sure how to go about it

Reply:
I don't think there is another way of doing it so it should be ok like that

Reply:
it gives a compile error if code is unreachable, refactored it